snapcraft/internal/meta/package_repository.py

```diff
@@ -141,24 +152,40 @@ def __init__(
         else:
             self.name = name
 
+        self.path = path
         self.suites = suites
         self.url = url
 
+        self._apt_repo = apt_repo
+
         self.validate()
 
     def install(self, keys_path: Path) -> bool:
+        if not self.path and not self.components and not self.suites:
+            suites = ["/"]
+        elif self.path:
+            # Suites denoting exact path must end with '/'.
+            path = self.path
+            if not path.endswith("/"):
+                path = path + "/"
```

Comment:
FWIW, this is also valid
```suggestion
                path += "/"
```

Reply:
Done.